frontend/components/course/Tile.tsx

```diff
@@ -2,17 +2,84 @@ import Image from "next/image";
 import Link from "next/link";
 import { FC } from "react";
 
+import { EnrollmentStatus_enum } from "../../__generated__/globalTypes";
+import { enrollmentStatusForCourse } from "../../helpers/courseHelpers";
 import { CourseList_Course } from "../../queries/__generated__/CourseList";
+import { CourseListWithEnrollments_Course } from "../../queries/__generated__/CourseListWithEnrollments";
 
 interface IProps {
-  course: CourseList_Course;
+  course: CourseList_Course | CourseListWithEnrollments_Course;
 }
 
+const colorForEnrollmentStatus = (
+  status: EnrollmentStatus_enum | "NOT_APPLIED"
+): string => {
+  if (
+    status === EnrollmentStatus_enum.APPLIED ||
+    status === EnrollmentStatus_enum.CONFIRMED
+  ) {
+    return "bg-edu-course-current";
+  }
+  if (
+    status === EnrollmentStatus_enum.REJECTED ||
+    status === EnrollmentStatus_enum.ABORTED
+  ) {
+    return "bg-gray-300";
+  }
+  if (status === EnrollmentStatus_enum.INVITED) {
+    return "bg-edu-course-invited";
+  }
+  return "";
```

Comment:
As you already mentioned: The `EXPIRED` status is  missing.

Reply:
Added it now that I have the data.